### PR TITLE
[Fix issue] Call completion closure on completed request

### PR DESCRIPTION
**Pull request by muukii**
Hello,

I found an issue that does not call completion closure on a completed request that fetches a set of products.
So firstly, I've put the code that fixes this issue with simple code.

If I correct, I'd like to talk to enhance my implementation.

Thanks for the awesome library.

### CODE REVIEW

File: Sources/SwiftyStoreKit/ProductsInfoController.swift [hunk 1]
```diff
@@ -69,9 +69,21 @@ class ProductsInfoController: NSObject {
...
+
+            let query = inflightRequests[productIds]!
+
+            if query.request.hasCompleted {
+                query.completionHandlers.forEach {
+                    $0(query.request.cachedResults!)
+                }
+            }
+
             return inflightRequests[productIds]!.request
```
Comment (muukii) [inline Sources/SwiftyStoreKit/ProductsInfoController.swift:87]: This section is the main part of this PR.
if the request has been completed, the completion closure that added won't be called.

Comment (muukii) [reply Sources/SwiftyStoreKit/ProductsInfoController.swift:87]: This causes an issue in purchasing operations.
If the developer retrieved a product before it. that operation would stop.

**Comment by bizz84** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests were not updated</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please target PRs to <code>develop</code> branch</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by muukii** [comment]
@bizz84 
Hi, This PR is still draft status. but I'd like to get your impression about this issue.
thanks

**Review by Sam-Spencer — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**